rmf_site_editor/src/sdf_loader.rs

```diff
@@ -35,7 +35,19 @@ pub struct SdfPlugin;
 
 impl Plugin for SdfPlugin {
     fn build(&self, app: &mut App) {
-        app.init_asset_loader::<SdfLoader>();
+        // Type registration is necessary to allow serializing the Scene that is loaded by this
+        // plugin
+        app.init_asset_loader::<SdfLoader>()
+            .register_type::<NameInSite>()
```

Comment:
Some of these components are used pretty broadly outside of the SdfPlugin, so it's not obvious to me that they should be grouped here exclusively.

What would you think of having a `RegisterCommonTypesPlugin` that has all of this in it, and which gets loaded by `SdfPlugin`?

Reply:
The main reason is that component registration is only needed when we are adding a component to a `Scene` and we want to serialize / deserialize such `Scene`, which is currently only done in the `SdfLoader` https://github.com/open-rmf/rmf_site/blob/983cd3c943e47924c5cf436e30becb3d4cb357c3/rmf_site_editor/src/sdf_loader.rs#L308
Trying to load an sdf with the `SdfLoader` _without_ registering all the types that it will add to the scene will result in a panic. If in the future we added a new component to the loader (i.e. parsing a new SDF property) without registering that will trigger a panic at the bevy level.

```
thread 'main' panicked at [...]/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_scene-0.12.1/src/scene_spawner.rs:430:35:
scene contains the unregistered type `rmf_site_format::misc::NameInSite`. consider reflecting it with `#[derive(Reflect)]` and registering the type using `app.register_type::<T>()`
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic in exclusive system `bevy_scene::scene_spawner::scene_spawner_system`!
Encountered a panic in system `bevy_app::main_schedule::Main::run_main`!
```

To me the strong coupling of registering the components in this plugin sends a message that these components _must_ be registered for this plugin to work, and if a new one was to be added in the future it also _must_ be registered. The moment the registration is done in another plugin somewhere else we might forget this and trigger panics when adding new components to the `SdfLoader`.

I think my recommendation here would be to keep it as is but refactor it into a `RegisterCommonTypesPlugin` as soon as we have another place where we serialize / deserialize scenes, at which point we might change the definition from "register everything that is needed by `SdfLoader`" to "register everything that the site editor needs" and make it a dependency of the `SitePlugin` instead.
That being said I noticed that all this coupling and potential panic info was all in my head and not in the code so I added a clearer note https://github.com/open-rmf/rmf_site/pull/239/commits/5275a35edce7c3351306e7550af6e144d4b1de09